### PR TITLE
User signup tests and documentation with chai

### PR DESCRIPTION
**Pull request by lnevado**
## Summary

Created the tests for signup and the documentation with chai.

## Trello Card

https://trello.com/c/LM9nuOg5/27-tests-y-documentaci%C3%B3n-de-sign-up

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const User = require('../models').user;
+
+exports.create = (req, res, next) => {
+  const user = req.body
```
Comment (igna92ts) [inline app/controllers/users.js:5]: validate the parameters

File: test/app.js [hunk 2]
```diff
@@ -19,12 +19,7 @@ beforeEach('drop tables, re-create them and populate sample data', done => {
         return `"public"."${table[0]}"`;
       })
       .join(', ');
-    return models.sequelize
-      .query(`TRUNCATE TABLE ${tableExpression} RESTART IDENTITY`)
-      .then(() => {
-        return dataCreation.execute();
-      })
-      .then(() => done());
+    return models.sequelize.query(`TRUNCATE TABLE ${tableExpression} RESTART IDENTITY`).then(() => done());
```
Comment (igna92ts) [inline test/app.js]: why did you change this?

File: test/users.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  server = require('../app'),
+  should = chai.should();
+
+chai.use(chaiHttp);
+describe('/users POST', () => {
+  it('should fail because email is missing', done => {
+    chai
+      .request(server)
+      .post('/users')
+      .send({
```
Comment (igna92ts) [inline test/users.js]: dont delegate error handling to the database, do it before it even reaches the controller logic, otherwise you may do a bunch of calculations before having it fail due to a saving error while you could have just validated it beforehand and avoided useless processing

File: test/users.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        firstName: 'John',
+        lastName: 'Smith',
+        email: 'john.smith@wolox.com.ar',
+        password: 'wolox1189'
+      })
+      .then(res => {
+        res.should.have.status(201);
```
Comment (igna92ts) [inline test/users.js:61]: validate that it was actually created in the database, the controller could just be returning 201 without doing anything else and this test would still pass. Import the db model and fetch the use and check that the inputs values match with the database values, including the encrypted password

File: test/users.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (igna92ts) [inline test/users.js:85]: why 201?

**Review by igna92ts — CHANGES_REQUESTED** [review]

File: app/controllers/users.js (hunk 1 above)
Comment (lnevado) [reply app/controllers/users.js:5]: Parameters are now validated

File: test/app.js (hunk 2 above)
Comment (lnevado) [reply test/app.js]: Created sample data and changed it back

File: test/users.js (hunk 4 above)
Comment (lnevado) [reply test/users.js:61]: Test now validates it was created correctly

File: test/users.js (hunk 5 above)
Comment (lnevado) [reply test/users.js:85]: The first sign up request with an email should be successful (201), the next one throws the error of a repeated email

File: app/controllers/users.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+const User = require('../models').user,
+  errors = require('../errors');
+
+exports.create = (req, res, next) => {
+  const user = req.body
+    ? {
```
Comment (igna92ts) [inline app/controllers/users.js]: there is no need to do this validation for existence in the body since you are already validating the parameters

**Review by igna92ts — CHANGES_REQUESTED** [review]

**Review by igna92ts — APPROVED** [review]
